src/p_mobj.c

```diff
@@ -890,6 +881,15 @@ mobj_t *P_SpawnMobj(fixed_t x, fixed_t y, fixed_t z, mobjtype_t type)
   mobj->z = z == ONFLOORZ ? mobj->floorz : z == ONCEILINGZ ?
     mobj->ceilingz - mobj->height : z;
 
+  // [AM] Do not interpolate on spawn.
+  mobj->interp = false;
+
+  // [AM] Just in case interpolation is attempted...
+  mobj->oldx = mobj->x;
+  mobj->oldy = mobj->y;
+  mobj->oldz = mobj->z;
+  mobj->oldangle = mobj->angle;
```

Comment:
https://github.com/fabiangreffrath/crispy-doom/blob/bf8cdbe493c40392be7667350b635a50c9cdb393/src/doom/p_mobj.c#L691-L711

Reply:
You mean, "adjust `mobj->z` first, then set `mobj->oldz = mobj->z`"? Sounds reasonable.